app/modules/gpio.c

```diff
@@ -87,59 +76,55 @@ static int lgpio_trig( lua_State* L )
 // Lua: mode( pin, mode, pullup )
 static int lgpio_mode( lua_State* L )
 {
-  unsigned mode, pullup = FLOAT;
-  unsigned pin;
+  unsigned pin = luaL_checkinteger( L, 1 );
+  unsigned mode = luaL_checkinteger( L, 2 );
+  unsigned pullup = luaL_optinteger( L, 3, FLOAT );
+  
+  luaL_argcheck(L, pin< NUM_GPIO && (mode!=INTERRUPT || pin>0), 1, "Invalid pin");
```

Comment:
I'll swap all of these out.  If you note, these are now inlined anyway so the compiler will emit the same compiled instruction sequence, but your suggestion is a better abstraction at a source level.